### ISSUE TITLE
[Request] "Subscribe" to searches

### ISSUE THREAD

**Issue by probonopd** [TeamNewPipe/NewPipe#844]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.

I would like to "subscribe" to certain topcis (searches), e.g., "all vidoes that appeared during the last week with for the search word AppImage". I would like to be able to save such searches, and optionally have NewPipe download all videos that appear in this search, either in the background, or manually triggered ("download now"), like in a podcast client.

NewPipe could keep track of the video IDs I have already watched in order not to download already-watched and deleted videos.

**Comment by TheAssassin** [comment]
Sounds useful, although I can't think of an efficient implementation. You'd have to repeat the search regularly, and look for newer results. That is likely a data usage issue. However, this might be a really useful feature to people who need to keep track of a specific tag etc. I'd like to see it implemented myself, I can think of use cases in which I might use it myself.

**Comment by theScrabi** [comment]
The only thing you could do is subscribe to search queues.

**Comment by probonopd** [comment]
> You'd have to repeat the search regularly, and look for newer results.

Yes. Could be done automatic (once a day, once a week, as any podcast client would do), and/or whenever the user initiates the process manually (this would actually be what I'd want to use myself). 

My use case is to download the results for a certain search before I leave the house for a long train ride.

**Comment by justanidea** [comment]
What about that today?

**Comment by probonopd** [comment]
> What about that today?

I'd appreciate this feature today (at least) as much as originally.

**Comment by theScrabi** [comment]
No progress yet.
The problem right now is that we get a ton if issues currently, but we are just so many devs.
For some things we really need help :)

**Comment by SameenAhnaf** [comment]
What about pinning searches instead??

**Comment by probonopd** [comment]
Can the results of all those pinned searches be shown in one combined view, ordered from most recent to least recent video?

**Comment by opusforlife2** [comment]
Good points. I wonder if the existing Channel Groups/Feeds code can be modified to do this as well.

**Comment by what-aboot** [comment]
> Sounds useful, although I can't think of an efficient implementation. You'd have to repeat the search regularly, and look for newer results. [/snip] 

If NP 
(A) saved the search as a channel, 
(B) prefix channel name with 'xnp' (or some such) 
(C) auto subscribed to it

This would do the search as the user opened (clicked on) the channel.

 It would only occur upon user activation, with user then prepared for data usage, not an ongoing automatic reloading. 

This would be efficient.

As search is now a channel, it can be managed by existing channel management process, incl delete.
